src/WebWindow.Native/WebWindow.Mac.NavigationDelegate.mm

```diff
@@ -0,0 +1,32 @@
+#import "WebWindow.Mac.NavigationDelegate.h"
+
+@implementation MyNavigationDelegate : NSObject
+
+- (void)webView:(WKWebView *)webView didStartProvisionalNavigation:(WKNavigation *)navigation;
+{
+    const char* uri = [webView.URL.absoluteString UTF8String];
+    int length = strlen(uri);
```

Comment:
Is `strlen` definitely correct here? Does it return the length in bytes or in characters? I'm concerned that for multibyte characters this would produce the wrong length and then `strncpy` would overflow the buffer.

Reply:
> Thanks for contributing this, @malscent! It looks like really good functionality.
> 
> Before I merge this, would you be willing to take a further pass through the PR and address the comments I've posted? Thanks again!

Yup.. I'll get those fixed up today.  Thanks for taking the time to review it!

Reply:
Ended up just factoring that out and finding a diffferent way to achieve the same thing without strlen or strncpy